rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -25,10 +26,28 @@
 
 from sawtooth_rest_api.protobuf.validator_pb2 import Message
 
-
 LOGGER = logging.getLogger(__name__)
 
 
+class _Backoff:
+    """Implements a simple exponential backoff mechanism.
+    """
+
+    def __init__(self):
+        self.num_retries_left = 3
+        self.min_interval = 0.1
+        self.max_interval = 1.0
+        self.interval = self.min_interval
+
+    async def do_backoff(self):
+        if not self.num_retries_left:
+            LOGGER.warning("No more retries left. Backoff terminated...")
+            raise DisconnectError()
```

Comment:
It's not likely that it's disconnected, here. We get notified of disconnection via the monitor socket.  This should be a timeout exception, which should effect the resulting http status - probably an HTTP status 408